tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/TlaType1.scala

```diff
@@ -6,27 +6,77 @@ import scala.collection.immutable.SortedMap
   * Trait for a type in Type System 1 as specified in
   * <a href="https://github.com/informalsystems/apalache/blob/unstable/docs/adr/002adr-types.md">ADR-002</a>.
   */
-sealed trait TlaType1
+sealed trait TlaType1 {
+  /**
+    * Compute the set of the names used in the type. These are names declared with VarT1.
+    * @return the set of variable names (actually, integers) that are used in the type.
+    */
+  def usedNames: Set[Int]
+
+  /**
+    * The interval [a, b) of variable numbers. This interval is used to quickly find, whether two types may
```

Comment:
`joinVarIntervals` is used in `EtcTypeChecker`. But since `EtcTypeChecker2` is a better implementation, I would remove the intervals altogether with `EtcTypeChecker`.

Reply:
Ok, cool. Do we want to do that in a followup to this pr? Or should I remove the interval here?

Reply:
I think it is ok to remove it right in this PR

Reply:
Done.